### PR TITLE
add wakeup for the Linak DPG1C controller

### PR DESCRIPTION
**Pull request by newAM**
CC #369

### CODE REVIEW

File: idasen/__init__.py [hunk 1]
```diff
@@ -238,6 +240,27 @@ def mac(self) -> str:
...
+        ...         await desk.wakeup()
+        >>> asyncio.run(example())
+        """
+        # https://github.com/rhyst/linak-controller/issues/32#issuecomment-1784055470
+        await self._client.write_gatt_char(_UUID_DPG, b"\x7F\x86\x00")
+        await self._client.write_gatt_char(
+            _UUID_DPG,
+            b"\x7F\x86\x80\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F\x10\x11",
+        )
+        await self._client.write_gatt_char(_UUID_COMMAND, COMMAND_WAKEUP)
```
Comment (kaml123) [inline idasen/__init__.py:262]: COMMAND_WAKEUP should be changed to _COMMAND_WAKEUP

File: idasen/__init__.py [hunk 2]
```diff
@@ -18,11 +18,13 @@
 _UUID_COMMAND: str = "99fa0002-338a-1024-8a49-009c0215f78a"
 _UUID_REFERENCE_INPUT: str = "99fa0031-338a-1024-8a49-009c0215f78a"
 _UUID_ADV_SVC: str = "99fa0001-338a-1024-8a49-009c0215f78a"
+_UUID_DPG: str = "99fa0010-338a-1024-8a49-009c0215f78a"
```
Comment (kaml123) [inline idasen/__init__.py:21]: this UUID is wrong it should be "99fa0011-338a-1024-8a49-009c0215f78a"

**Review by kaml123 — COMMENTED** [review]
I also propose to add self.wakeup() inside **async def connect(self):**

**Comment by newAM** [comment]
@abmantis if you have time your input here would be great since you maintain the HASS component.  Do you know of any problems with this approach?

I tried this with my unmodified desk/controller and found no issues sending the wakeup command, but it's always good to double check :)

**Comment by abmantis** [comment]
Sorry for the delay in replying @newAM ! I will try to test it over the weekend. 
Is it possible to bump the major version of this lib when releasing this? That way, I can delay HA from using it. I've had quite a few reports from users regarding bluetooth instability (not related to this lib) and have an improvement that I want to get into the next release of HA. I would prefer to avoid adding this as a possible cause of extra unknown issues before the current ones are solved :sweat_smile: 

If you are fine with bumping the major version of the lib, than feel free to merge this as it will not affect HA for now! Later, if I find any issues during tests we can work on them :)

**Comment by datbilling** [comment]
Hi @abmantis, I was just wondering if you had time yet to test this to see if it can be added into HA?

**Comment by abmantis** [comment]
> Hi @abmantis, I was just wondering if you had time yet to test this to see if it can be added into HA?

Yes, it worked fine on my desk (not DPG1C) and should be released with HA 2024.2!

**Comment by datbilling** [comment]
> > Hi @abmantis, I was just wondering if you had time yet to test this to see if it can be added into HA?
> 
> 
> 
> Yes, it worked fine on my desk (not DPG1C) and should be released with HA 2024.2!

Ok perfect, thank you!

**Pull request merged**